docs/02.API-REFERENCE.md

```diff
@@ -169,6 +169,28 @@ typedef struct
 } jerry_property_descriptor_t;
 ```
 
+## jerry_heap_stats_t
+
+**summary**
+
+Description of JerryScript heap memory stats.
+It is for memory profiling.
+
+**Prototype**
+
+```c
+typedef struct
+{
+  size_t version /**< the version of the stats struct */
+  size_t size; /**< heap total size */
+  size_t allocated_bytes; /**< currently allocated bytes */
+  size_t peak_allocated_bytes; /**< peak allocated bytes */
+  size_t waste_bytes; /**< bytes waste due to blocks filled partially */
+  size_t peak_waste_bytes; /**< peak wasted bytes */
```

Comment:
It's fbmrk's idea. What do you think? @fbmrk 
Personally, I also agree that the `waste` is much more impl-specific than `allocated` and `total`